scripts/release/goa_prepare/goa_submission.pl

```diff
@@ -274,59 +277,118 @@ sub get_reaction_id {
 
 sub get_proteins_from_catalyst_activity {
     my $parameters = shift;
+
+    my $logger = get_logger(__PACKAGE__);
     
     my $catalyst_activity = $parameters->{'catalyst_activity'};
     my $ontology_letter = $parameters->{'ontology_letter'};
     
-    my @errors = check_catalyst_activity($catalyst_activity);
-    if (@errors) {
-        print(join("\n", @errors) . "\n");
+    my @reasons_to_exclude = check_catalyst_activity($catalyst_activity, $ontology_letter);
+    if (@reasons_to_exclude) {
+        $logger->info(join("\n", @reasons_to_exclude) . "\n");
         return;
-    }          
+    }
     
-    if ($catalyst_activity->activeUnit->[0] && $ontology_letter eq 'F') {
-        return get_proteins_from_physical_entity($catalyst_activity->activeUnit->[0]);
+    if ($ontology_letter eq 'F') {
+        return get_proteins_for_molecular_function(
+            $catalyst_activity->activeUnit->[0] ?
+            $catalyst_activity->activeUnit->[0] :
+            $catalyst_activity->physicalEntity->[0]
+        );
+    } elsif ($ontology_letter eq 'P') {
+        return get_proteins_for_biological_process($catalyst_activity->physicalEntity->[0]);
     }
-    return get_proteins_from_physical_entity($catalyst_activity->physicalEntity->[0]);
 }
 
 sub check_catalyst_activity {
     my $catalyst_activity = shift;
-    
-    my @errors;
+    my $ontology_letter = shift;
     
     my $catalyst_id = $catalyst_activity->displayName . ' (' . $catalyst_activity->db_id . ')';
     my $physical_entity = $catalyst_activity->physicalEntity->[0];
     my @active_units = @{$catalyst_activity->activeUnit};
+
+    my @reasons_to_exclude;
+    push @reasons_to_exclude, "No physical entity: $catalyst_id"
+        if (!$physical_entity);
+    push @reasons_to_exclude, "No compartment for physical entity: $catalyst_id"
+        unless $physical_entity && $physical_entity->compartment->[0];
```

Comment:
This block could be slightly refactored to make it more readable, by using an 'if-else' if block that checks for existence of PE, and then for compartment

Reply:
Good point -- I'll refactor that